R-packages/covidcast/R/covidcast.R

```diff
@@ -642,7 +645,14 @@ covidcast_signals <- function(data_source, signal,
 #' @importFrom utils read.csv
 #' @export
 covidcast_meta <- function() {
-  meta <- .request("covidcast_meta", list(format = "csv"))
+  # use cached metadata whenever possible
+  META_RESPONSE <- ifelse(
+    is.na(META_RESPONSE),
+    .request("covidcast_meta", list(format = "csv"), raw = TRUE),
+    httr::rerequest(META_RESPONSE)
+  )
```

Comment:
This probably should be:

```suggestion
  META_RESPONSE <<- if (is.na(META_RESPONSE)) {
      .request("covidcast_meta", list(format = "csv"), raw = TRUE)
    } else  {
      httr::rerequest(META_RESPONSE)
  }
```

The important change is `<<-`, so we write to the package environment instead of the function local environment. Using `ifelse()` works here because it lazily evaluates its arguments, so we don't evaluate both eagerly, but that's non-obvious enough that I had to look it up. An `if` expression also seems nice.